### PR TITLE
Guide signs - parsing and setting signs only

### PR DESCRIPTION
**Pull request by gknisely**
# Issue

Added logic to parse destination info on non-link edges.  This edges will have signs but they will be guide signs and used at intersections for example.  This PR only parses and saves the info and does not send the results to Odin.  That PR is to come along with more tests.

## Tasklist

 - [ ] Review - you must request approval to merge any PR to master
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [ ] Update relevant [documentation](docs/README.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/mjolnir/graphbuilder.cc [hunk 1]
```diff
@@ -1250,7 +1259,11 @@ std::vector<SignInfo> GraphBuilder::CreateExitSignInfoList(const OSMNode& node,
     std::vector<std::string> branch_streets =
         GetTagTokens(osmdata.name_offset_map.name(way.destination_street_index()));
     for (auto& branch_street : branch_streets) {
-      exit_list.emplace_back(Sign::Type::kExitBranch, false, branch_street);
+      if (!link) {
```
Comment (dgearhart) [inline src/mjolnir/graphbuilder.cc]: There are exit signs that are not on links - for example: https://www.openstreetmap.org/way/6009977#map=16/39.3222/-76.5308

Comment (gknisely) [reply src/mjolnir/graphbuilder.cc]: a link = turn channel or ramp

File: src/sif/autocost.cc [hunk 2]
```diff
@@ -7,7 +7,6 @@
 #include "baldr/nodeinfo.h"
 #include "midgard/constants.h"
 #include "midgard/util.h"
-#include "sif/dynamiccost.h"
```
Comment (kevinkreiser) [inline src/sif/autocost.cc]: why are these diffs here. this branch isnt intended to change any of this. @PureW any clues as to how this is basically the inverse of your changes?

Comment (kevinkreiser) [reply src/sif/autocost.cc]: i know rewriting git history can cause stuff like this but we havent ever rewritten history on master, so how can this be?

File: src/sif/autocost.cc [hunk 3]
```diff
@@ -470,7 +497,7 @@ Cost AutoCost::TransitionCost(const baldr::DirectedEdge* edge,
     // Still want to add a penalty so routes avoid high cost intersections.
     float seconds = turn_cost * edge->stopimpact(idx);
     // Apply density factor penality if there isnt traffic on this edge or youre not using traffic
-    if (!edge->has_flow_speed() || flow_mask_ == 0)
+    if (!edge->has_flow_speed() && flow_mask_ != 0)
```
Comment (kevinkreiser) [inline src/sif/autocost.cc]: and this is my change, which was squash merged (ie a rebase of my branch and then a merge)

**Comment by kevinkreiser** [comment]
sadly this branches commit history got mixed up in such a way that it undoes changes in master. ive made a new branch where we should have just the changes from this branch: https://github.com/valhalla/valhalla/pull/2096

**Pull request closed without merging**